csrc/multidevice/executor.cpp

```diff
@@ -196,6 +196,7 @@ void MultiDeviceExecutor::postCommunication(SegmentedGroup* group) {
     auto work = communication->post(comm_);
     if (work) {
       work->wait();
+      communication->relayoutOutputTensor();
```

Comment:
This call should be wrapped into "work->wait" because we shouldn't expect the user of these instances to always call `relayoutOutputTensor` right after `wait`

Reply:
We should add `requiresRelayoutOutputTensor` no ?